File: app/src/main/java/com/demo/AppV4.java
package com.demo;

import com.demo.cg.IntegerAdder;
import com.demo.util.SimpleMethodInterceptor;
import net.sf.cglib.core.DebuggingClassWriter;
import net.sf.cglib.proxy.Enhancer;
import net.sf.cglib.proxy.MethodInterceptor;

public class AppV4 {
    public static void main(String[] args) {
        System.setProperty(DebuggingClassWriter.DEBUG_LOCATION_PROPERTY, "./generated");
        MethodInterceptor interceptor = new SimpleMethodInterceptor();

        IntegerAdder calculator =
                (IntegerAdder) Enhancer.create(IntegerAdder.class, interceptor);
        calculator.add(1, 2);
    }
}

File: app/src/main/kotlin/com/demo/AppV4.kt
package com.demo

import com.demo.cg.IntegerAdder
import com.demo.util.SimpleMethodInterceptor
import net.sf.cglib.core.DebuggingClassWriter
import net.sf.cglib.proxy.Enhancer

object AppV4 {

    private const val OUTPUT_DIR = "./generated/v4"

    @JvmStatic
    fun main(args: Array<String>) {
        System.setProperty(DebuggingClassWriter.DEBUG_LOCATION_PROPERTY, OUTPUT_DIR)
        val methodInterceptor = SimpleMethodInterceptor()
        val calculator =
            Enhancer.create(IntegerAdder::class.java, methodInterceptor) as IntegerAdder
        val result = calculator.add(1, 2)
        println("result: $result")
    }
}